### PR TITLE
Implement derive macro for Aggregate

### PR DESCRIPTION
**Pull request by ffuugoo**
Example:

```rust
#[derive(Default, Aggregate)]
#[aggregate(type = "inferred.id.aggregate")]
struct InferredIdAggregate {
    id: i32,
    value: String,
}

#[derive(Default, Aggregate)]
#[aggregate(type = "explicit.id.aggregate")]
struct ExplicitIdAggregate {
    #[aggregate(id)]
    explicit_id: i32,
    value: String,
}

#[derive(Default, Aggregate)]
#[aggregate(type = "tuple.struct.aggregate")]
struct TupleStructAggregate(#[aggregate(id)] i32, String);
```

### CODE REVIEW

**Comment by ffuugoo** [comment]
### FCM

```
Implement derive macro for Aggregate

- reorganize crate implementations to maximize code reuse

Additionally:
- fix docstrings for some event-family derives implementation
```

File: cqrs-codegen/impl/src/aggregate.rs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            .find(|(_, field)| {
+                match &field.ident {
+                    Some(ident) => ident == "id",
+                    None => false,
+                }
+            });
+    }
+
+    id
+        .map(|(index, field)| {
```
Comment (tyranron) [inline cqrs-codegen/impl/src/aggregate.rs]: You definitely should apply `cargo +nightly fmt -p cqrs-codegen-impl` before requesting review.

**Comment by tyranron** [comment]
@ffuugoo please, don't be in hurry. It's better to make a task slower, but with more quality.

File: cqrs-codegen/impl/src/lib.rs [hunk 2]
```diff
@@ -33,7 +40,8 @@ pub fn expand<TS: From<TokenStream>>(
     }
 }
 
-export!(aggregate_event_derive);
-export!(event_derive);
-export!(registered_event_derive);
-export!(versioned_event_derive);
+export!{aggregate::derive as aggregate_derive}
```
[the remark this answers is not in the document]
Comment (ffuugoo) [reply cqrs-codegen/impl/src/lib.rs]: I've hit compiler diagnostic about "macro generating items must use braces". I'll be more careful with it and will re-check it before reviews from now on.

**Pull request merged**